File: Z_practise/302_allbinary.java
class Solution {
    
    private void rec(String num, int i, ArrayList<String> ans){
        if(i==0){
            ans.add(num);
            return;
        }
        
        rec(num+"0", i-1, ans);
        rec(num+"1", i-1, ans);
    }
    
    public ArrayList<String> binstr(int n) {
        // code here
        ArrayList<String> ans = new ArrayList<>();
        rec("", n, ans);
        return ans;
    }
}

File: Z_practise/302_allbinary.py
class Solution:
    
    def __init__(self):
        self.ans = []
    
    def rec(self, i, str):
        if i==0:
            self.ans.append(str)
            return
        
        self.rec( i-1, str+"0")
        self.rec( i-1, str+"1")
    
    def binstr(self, n):
        # code here
        self.rec( n, "")
        return self.ans
        